### PR TITLE
network: nexthop: add OnLink= setting

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18589.

### CODE REVIEW

File: man/systemd.network.xml [hunk 1]
```diff
@@ -1340,6 +1340,15 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
             to <literal>ipv4</literal>.</para>
           </listitem>
         </varlistentry>
+        <varlistentry>
+          <term><varname>OnLink=</varname></term>
+          <listitem>
+            <para>Takes a boolean. If set to true, the kernel does not have to check if the gateway is
+            reachable directly by the current machine (i.e., the kernel does not need to check if the
```
Comment (bluca) [inline man/systemd.network.xml]: A bit repetitive, maybe "(i.e., attached to the local network)

File: man/systemd.network.xml [hunk 2]
```diff
@@ -1340,6 +1340,15 @@ IPv6Token=prefixstable:2002:da8:1::</programlisting></para>
...
+            gateway is attached to the local network), so that we can insert the nexthop in the kernel
+            table without it being complained about. Defaults to <literal>no</literal>.</para>
```
Comment (bluca) [inline man/systemd.network.xml]: It's also enabled automatically in one case, no? So maybe "Defaults to no, unless Gateway= is set without a static address configured, in which case it is enabled"

File: src/network/networkd-nexthop.c [hunk 3]
```diff
@@ -549,6 +556,14 @@ static int nexthop_section_verify(NextHop *nh) {
                 /* When no Gateway= is specified, assume IPv4. */
                 nh->family = AF_INET;
 
+        if (nh->onlink < 0 && in_addr_is_set(nh->family, &nh->gw) &&
```
Comment (bluca) [inline src/network/networkd-nexthop.c:559]: Maybe add a code comment here explaining why it is enabled automatically in this one case, for extra clarity for readers

File: man/systemd.network.xml (hunk 2 above)
Comment (yuwata) [reply man/systemd.network.xml]: I'd like to change the condition in the future. So, I'd like to not mention that. If no address is configured, then we emit a warning. So, I hope it is enough now.

Currently, we only checks static addresses. But any dynamic addresses should be also OK. However, unfortunately, we have no way to configure static routes with dynamic addresses now. I'd like to improve such situation in the future.

**Comment by yuwata** [comment]
@bluca Thank you for the review. All comments except for about the default value sentence are addressed. PTAL.

File: man/systemd.network.xml (hunk 2 above)
Comment (bluca) [reply man/systemd.network.xml]: Ok, makes sense

**Pull request merged**